src/api/entitlements.js

```diff
@@ -373,18 +384,29 @@ export class Entitlement {
     const products = json['products'] || [];
     const subscriptionToken = json['subscriptionToken'];
     let subscriptionTokenContents;
+    const timestampJson = json['subscriptionTimestamp'];
+    let subscriptionTimestamp;
```

Comment:
Could you move these variable declarations down so they're just above the new try/catch block? I think that might make the code a little simpler, since the first try/catch block doesn't need these new variables